Added missing/more content

Opisek opened the pull request:
- Added missing elements: concave and convex resistors, zener and tunnel diodes
- Added a mention of how to calculate the i_q vector in the exam
- Added information about duality conversion of reactive elements
- Added more mathematical tools

Opisek committed:
fix wrong symbol for "much smaller than"

Opisek committed:
fix wrong font for "det"

Opisek committed:
fix inconsistent usage of mathcal

Opisek committed:
fix wrong font for "lin"

Opisek committed:
fix more lacking math mode

Opisek committed:
revert mistakenly changed tabulation

Opisek committed:
Merge branch 'latex4ei:master' into master

Opisek committed:
add function name conventions for u, q, i, Φ

Opisek committed:
dualwandlung von reaktanzen

Opisek committed:
mehr mathe

Opisek committed:
add more elements from the lecture

Opisek committed:
mention of how to calculate i_q

Opisek committed:
fix a typo

Opisek committed:
fix wrong function names

dariusptrs committed:
fix workflow

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -867,6 +883,103 @@
+        \subsection{Differentiation}
+        \begin{tabular}{l|l}
+            $f(x)$        & $\dot f(x)=\frac{\mathrm d}{\mathrm dx}f(x)$ \\\hline
+            $xa$ & $1$ \\
```
This is wrong, I would also move the constant to the front.
```suggestion
            $ax$ & $a$ \\
```

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -867,6 +883,103 @@
+            $f(x)$        & $\dot f(x)=\frac{\mathrm d}{\mathrm dx}f(x)$ \\\hline
+            $xa$ & $1$ \\
+            $x^a$ & $a\cdot x^{a-1}$ \\
+            $e^{x}$ & $a\cdot e^{x}$ \\
```
This is also wrong
```suggestion
            $e^{ax}$ & $a\cdot e^{x}$ \\
```

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -867,6 +883,103 @@
+            $x^a$ & $\frac{1}{a+1}x^{a+1}$ \\
+            $e^{x}$ & $e^{x}$ \\
+            $\ln(x)$ & $x\ln(x)-x$ \\
+            $\frac{1}{x}$ & $\ln(x)$ \\
```
I would suggest adding absolute value bars, even though it is probably not necessary.
```suggestion
            $\frac{1}{x}$ & $\ln\left|x\right|$ \\
```

Opisek replied:
I had them at first, then I saw the TA explicitly write the derivative without the absolute value during an exercise. I agree that putting them back in is the correct choice, though.

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -867,6 +883,103 @@
+    \sectionbox{
+        \subsection{Trigonometrie}
+        $\cos(x) = \cos(-x)$\\
+        $\sin(x) = -\sin(x)$\\
```
```suggestion
        $\sin(-x) = -\sin(x)$\\
```

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -303,6 +288,23 @@
+        \parbox{1.3cm}{ Norator      \\[5pt] \includegraphics{./img/symb/symb_norator.pdf} } & \parbox{2.5cm}{$u = \mathrm{beliebig}$ \\ $i=\mathrm{beliebig}$ \\[0.5em] u, sl, a, dual zu sich selbst} & \parbox{1.8cm}{\includegraphics{./img/char/char_norator.pdf} } \\ \mrule
+        \parbox{1.3cm}{ Widerstand   \\[5pt] \includegraphics{./img/symb/symb_resistor.pdf} }& $\begin{array}{@{}cc} u = R \cdot i \\ i = G \cdot u \end{array} \quad G = \frac{1}{R}$ & \parbox{1.8cm}{\includegraphics{./img/char/char_resistor.pdf} } \\ \mrule
+        \parbox{1.3cm}{ \shortstack{Konkaver\\Widerstand}   \\ \includegraphics{./img/symb/symb_resistor_concave.png} }& $i=\begin{cases}0&\text{, }u\le U\\ G(u-U)&\text{, }u>U\end{cases}$ & \parbox{1.8cm}{\includegraphics[width=1.8cm]{./img/char/char_resistor_concave.png} } \\ \mrule
+        \parbox{1.3cm}{ \shortstack{Konvexer\\Widerstand}   \\ \includegraphics{./img/symb/symb_resistor_convex.png} }& \shortstack{Dual zum\\\textbf{konkaven Widerstand}} & \parbox{1.8cm}{\includegraphics[width=1.6cm]{./img/char/char_resistor_convex.png} } \\ \mrule
```
Add definition for dual element
```suggestion
        \parbox{1.3cm}{ \shortstack{Konvexer\\Widerstand}   \\ \includegraphics{./img/symb/symb_resistor_convex.png} }& \shortstack{ \shortstack{Dual zum\\\textbf{konkaven Widerstand}} \\ $u=\begin{cases}0&\text{, }i\le I\\ R(i-I)&\text{, }i>I\end{cases}$ }& \parbox{1.8cm}{\includegraphics[width=1.6cm]{./img/char/char_resistor_convex.png} } \\ \mrule

```

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -303,6 +288,23 @@
+        \parbox{1.3cm}{ ideale Diode \\[5pt] \includegraphics{./img/symb/symb_diode.pdf} }& $\begin{array}{@{}ll} u = 0 & \text{falls } i > 0 \\ i = 0 & \text{falls } u < 0\end{array}$ & \parbox{1.8cm}{\includegraphics{./img/char/char_diode_ideal.pdf} } \\ \mrule
+        \parbox{1.3cm}{ \shortstack{reale Diode\\pn Diode}  \\[5pt] \includegraphics{./img/symb/symb_diode_real.pdf} }& $\begin{array}{@{}ll} u_D = u_T \cdot \ln \left(\frac{i_D}{I_S} + 1 \right) \\[0.7em] i_D = I_S \cdot \left( \exp \left(\frac{u_D}{U_T}\right) -1 \right) \end{array}$ & \parbox{1.8cm}{\includegraphics{./img/char/char_diode_real.pdf} } \\ \mrule
+        \parbox{1.3cm}{ Photodiode   \\ \includegraphics{./img/symb/symb_diode_photo.pdf} }& $i = I_{\ir S} \left( e^{\frac{u_{\ir D}}{U_{\ir T}}} -1 \right) - i_L$ & \parbox{1.8cm}{\includegraphics{./img/char/char_diode_photo.pdf} } \\ \mrule
+        \parbox{1.3cm}{ Zenerdiode   \\ \includegraphics{./img/symb/symb_diode_zener3.png} }& Durchbruch bei $u<U_Z\le 0$: im \textbf{negativen Bereich} leitend & \parbox{1.8cm}{\includegraphics[width=1.8cm]{./img/char/char_diode_zener.png} } \\ \mrule
```
```suggestion
        \parbox{1.3cm}{ Zenerdiode   \\ \includegraphics{./img/symb/symb_diode_zener3.png} }&\shortstack{ Durchbruch bei $u=U_Z$:\\ für $u\leq U_z$ stark leitend} & \parbox{1.8cm}{\includegraphics[width=1.8cm]{./img/char/char_diode_zener.png} } \\ \mrule
```

dariusptrs replied:
Not sure about this topic, maybe you can share your source? To my understanding the Zenerdiode really begins to conduct for |u|>|Uz| and before that there is almost no current going through.

dariusptrs replied:
Do you have a screenshot/reference from the script for `Durchbruch bei $u<U_Z\le 0$: im \textbf{negativen Bereich} leitend`? 
My suggestions come from what I found online about this topic.

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -636,6 +638,13 @@
                          & \vdots & \\}$ & \hspace{-2em}\parbox{3cm}{\includegraphics[scale=0.15]{./img/nodevoltageanalysis/vccs_a_y_gnd.png} }                                                                                                                                         \\
     \end{tabular}\\
     \normalsize
+    \subsubsection{Aufstellen des Knottenstromquellenvektors}
```
```suggestion
    \subsubsection{Aufstellen des Knoten-Stromquellenvektors}
```

dariusptrs replied:
Typo in heading

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -636,6 +638,13 @@
     \normalsize
+    \subsubsection{Aufstellen des Knottenstromquellenvektors}
+    Bei dem Vektor $\vec i_{q}$ werden die \textbf{konstanten Quellenströme} eingetragen.
+    Anders als üblicherweise, trägt man in der $i$-ten Zeile den Strom $I_0$, der aus dem Knoten \textbf{rausfließt}, \textbf{negativ} ein und den Strom $I_0$, der in den Knoten \textbf{hineinfließt}, \textbf{positiv} ein.\\
```
We could improve clarity here
```suggestion
    Anders als üblicherweise, trägt man in der jeweiligen Zeile des Knotens \textbf{herausfließende} Ströme \textbf{negativ} ein und \textbf{hineinfließende} Ströme \textbf{positiv} ein.\\
```

dariusptrs replied:
We can also use your definition if you really like it more.

dariusptrs replied:
This section still needs to be resolved.

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -636,6 +638,13 @@
+    Bei dem Vektor $\vec i_{q}$ werden die \textbf{konstanten Quellenströme} eingetragen.
+    Anders als üblicherweise, trägt man in der $i$-ten Zeile den Strom $I_0$, der aus dem Knoten \textbf{rausfließt}, \textbf{negativ} ein und den Strom $I_0$, der in den Knoten \textbf{hineinfließt}, \textbf{positiv} ein.\\
+
+    Falls die Quelle auf der einen Seite mit der Masse verbunden ist, und der Strom in den Knoten hineinfließt, ergibt sich beispielhaft folgender Knotenstromquellenvektor:
```
I think it is better to include a definition instead of an example
```suggestion
    Ist eine Quelle auf einer Seite mit dem Masseknoten verbunden, wird nur der Wert für den gegenüberliegenden (nicht mit Masse verbundenen) Knoten im Stromquellenvektor eingetragen.
```

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -636,6 +638,13 @@
+    Anders als üblicherweise, trägt man in der $i$-ten Zeile den Strom $I_0$, der aus dem Knoten \textbf{rausfließt}, \textbf{negativ} ein und den Strom $I_0$, der in den Knoten \textbf{hineinfließt}, \textbf{positiv} ein.\\
+
+    Falls die Quelle auf der einen Seite mit der Masse verbunden ist, und der Strom in den Knoten hineinfließt, ergibt sich beispielhaft folgender Knotenstromquellenvektor:
+    $\vec i_{q} = \mat{0 &\cdots &I_0 &\cdots &0}^T$
```
```suggestion
```

Opisek replied:
Did you mean to also delete the following?

```
Falls die Quelle auf der einen Seite mit der Masse verbunden ist, und der Strom in den Knoten hineinfließt, ergibt sich beispielhaft folgender Knotenstromquellenvektor:
```

dariusptrs replied:
You can leave it in but I would suggest to remove the `\cdots` and give a little bit more context. 
E.g. "Fließt ein Quellenstrom $I_0$ vom Masseknoten zu Knoten 2 ergibt sich folgender Stromquellenvektor:
`$\vec i_{q} = \mat{0 &I_0 &0}^T$`"

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -778,8 +787,8 @@
     \textbf{Energie (nicht linearer Fall)}:\\
-    - Kapazitiv: $W_C(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{q_1}^{q_2} \! u(q)\, \mathrm dq$\\
-    - Induktiv: $W_C(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{\Phi_1}^{\Phi_2} \! i(\Phi)\, \mathrm d\Phi$\\
+    - Kapazitiv: $W_C(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{q_1}^{q_2} \! \chi(q)\, \mathrm dq$\\
```
Even though the previous notation was also correct. I support using your notation and added the relation between `\chi` and `u`
```suggestion
    - Kapazitiv: $W_C(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{q_1}^{q_2} \! \chi(q)\, \mathrm dq$, $u=\chi(q)$\\
```

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -778,8 +787,8 @@
-    - Kapazitiv: $W_C(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{q_1}^{q_2} \! u(q)\, \mathrm dq$\\
-    - Induktiv: $W_C(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{\Phi_1}^{\Phi_2} \! i(\Phi)\, \mathrm d\Phi$\\
+    - Kapazitiv: $W_C(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{q_1}^{q_2} \! \chi(q)\, \mathrm dq$\\
+    - Induktiv: $W_L(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{\Phi_1}^{\Phi_2} \! \lambda(\Phi)\, \mathrm d\Phi$\\
```
add relation between `\lambda` and `\Phi`
```suggestion
    - Induktiv: $W_L(t_1, t_2) = \int_{t_1}^{t_2} \! u(t)i(t)\, \mathrm dt = \int_{\Phi_1}^{\Phi_2} \! \lambda(\Phi)\, \mathrm d\Phi$, $i=\lambda(\Phi)$\\
```

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -867,6 +883,103 @@
+        % Credit:
+        % https://tex.stackexchange.com/questions/637299/draw-trigonometric-circle-with-tikz-package
+        \definecolor{col_sectionbox}{RGB}{251, 251, 251} % for some reason the color macros from the template don't work
+        \textbf{Wichtige Sinus- und Cosinuswerte} $(\cos x, \sin x)$\\
```
We already have a table for important sin and cos values, but I guess we can also add this.

dariusptrs reviewed (changes requested):
Thanks for your contributions!

I've added some suggestions to clarify a few sections and pointed out a few small mistakes that need to be resolved.

Opisek committed:
wrong linear derivative

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
missing constant

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
abs val for 1/x derivative

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
missing minus

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
missing minus

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
more consice explanation

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
dual element definition

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
add chi relation

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
add phi relation

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
add cramer'sche regel

Opisek committed:
fix missing backslashes

Opisek committed:
add section about complex numbers

Opisek committed:
matrix/vector styling issues

Opisek committed:
add duality symbol

Opisek committed:
source conversion

Opisek commented:
Thanks for the rigorous review @dariusptrs 

I corrected the mistakes you found, but didn't commit one of your changes yet (see my response in the unresolved suggestion).

I also added some information about complex numbers that might help with "Komplexe Wechselstromrechnung", a graphic for source conversion and fixed a few mistakes / imprecisions I encountered.

Please check again.

Opisek committed:
kreisfrequenz

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -867,6 +902,127 @@
+    }
+    \sectionbox{
+        \subsection{Komplexe Zahlen}
+        \textbf{Karthesische Darstellung}: $z = a + j b$\\
```
```suggestion
        \textbf{Kartesische Darstellung}: $z = a + j b$\\
```

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -867,6 +902,127 @@
+
+        \textbf{Erweitern mit Komplex Konjugiertem}:\\
+        $$
+        \frac{x}{a+jb} = \frac{x(a-jb)}{a^2+b^2} = \frac{ax}{a^2+b^2} - j\frac{bx}{a^2+b^2}
```
This is great!

dariusptrs reviewed (commented):
Thanks for committing the fixes and adding even more useful content!

See my messages in the open discussions and let me know what you think.

Opisek committed:
typo

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek committed:
typo

Opisek committed:
add example

Opisek commented:
The zener diode stuff is taken straight from the lecture (corresponds to 3.4.3 in the lecture script).

Everything other discussion should be resolved by now, if I hadn't missed anything.

Personally I would also love to see some more explanation about "Großsignal ESB / Kleinsignal ESB", but I'm not going to add that in this PR.

Opisek committed:
fix typo

dariusptrs committed:
fix latexdiff fail

dariusptrs committed:
fix latexdiff fail 2

dariusptrs commented on Schaltungstechnik.tex:
```diff
@@ -867,6 +906,127 @@
+        \textbf{Polarkoordinaten}: $z = r \cdot e^{j \varphi}$\\
+
+        Umrechnung:\\
+        $r = \sqrt{a^2 + b^2}$, $\varphi = \arctan\left(\frac{b}{a}\right)$\\
```
```suggestion
        $r = \sqrt{a^2 + b^2}$, $\varphi = \begin{cases}
                        \arctan\frac{b}{a}     & a>0 \\
                        \arctan\frac{b}{a}+\pi & a<0
                    \end{cases}$\\
```

dariusptrs reviewed (commented):
Great, I suggested one more thing in your math part and there are still two other open conversations.

Opisek committed:
case distinction for polar angles

Co-authored-by: Darius Peters <38152878+dariusptrs@users.noreply.github.com>

Opisek commented:
For the Knotenstromquellenvektor, I'm fine with both definitions. Feel free to change it to yours if you prefer it, but in that case, please keep the example, so one can also infer which vector row corresponds to which vertex.

For the zener diode:

![image](https://github.com/user-attachments/assets/271b0a5f-9d7b-4c7b-a2ef-6e22f31070c6)

dariusptrs committed:
change definition for Knoten-Stromquellenvektor

dariusptrs committed:
update Zenerdiode
